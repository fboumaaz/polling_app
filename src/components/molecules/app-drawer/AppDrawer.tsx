// copied from  https://mui.com/components/drawers/#mini-variant-drawer

import * as React from 'react';
import MenuIcon from '@mui/icons-material/Menu';
import Box from '@mui/material/Box';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import CssBaseline from '@mui/material/CssBaseline';
import Typography from '@mui/material/Typography';
import IconButton from '@mui/material/IconButton';

interface AppDrawerProps {
  children: JSX.Element;
}

const AppDrawer = ({ children }: AppDrawerProps) => (
  <Box sx={{ display: 'flex' }}>
    <CssBaseline />
    <AppBar position="fixed">
      <Toolbar>
        <IconButton
          edge="start"
          sx={{
            mr: '36px',
          }}
        >
          <MenuIcon />
        </IconButton>
        <Typography variant="h5" noWrap>
          Voting poll
        </Typography>
      </Toolbar>
    </AppBar>
    <Box
      component="main"
      sx={{
        flexGrow: 1,
        p: 5,
      }}
    >
      {children}
    </Box>
  </Box>
);
export default AppDrawer;
