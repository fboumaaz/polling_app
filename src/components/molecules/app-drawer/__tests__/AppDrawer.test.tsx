import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import AppDrawer from "../AppDrawer";
import { Box, AppBar, Toolbar } from "@mui/material";

configure({ adapter: new Adapter() });

describe("home page", () => {
  const wrapper = shallow(
    <AppDrawer>
      <div />
    </AppDrawer>
  );

  it("renders a Box", () => {
    expect(wrapper.find(Box)).toHaveLength(2);
  });

  it("renders a MuiAppBar", () => {
    expect(wrapper.find(AppBar)).toHaveLength(1);
  });

  it("renders a Toolbar", () => {
    expect(wrapper.find(Toolbar)).toHaveLength(1);
  });

  it("renders snapshots", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
