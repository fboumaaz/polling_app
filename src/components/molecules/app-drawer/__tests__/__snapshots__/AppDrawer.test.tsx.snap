// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page renders snapshots 1`] = `ShallowWrapper {}`;
